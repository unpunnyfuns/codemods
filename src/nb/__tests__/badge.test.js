import { describe, expect, it } from 'vitest'
import { DEFAULT_TEST_OPTIONS, testTransform } from '../../__tests__/test-helper.js'

describe('nb/badge', () => {
  it('migrates Badge component', () => {
    const output = testTransform('nb/badge', 'nb/badge', 'tsx', {
      sourceImport: DEFAULT_TEST_OPTIONS.sourceImport,
      targetImport: '@target/components/Badge',
      tokenImport: DEFAULT_TEST_OPTIONS.tokenImport,
    })
    expect(output).toMatchSnapshot()
  })
})
