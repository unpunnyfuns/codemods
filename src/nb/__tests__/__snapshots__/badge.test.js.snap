// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nb/badge > migrates Badge component 1`] = `
"import { Badge } from 'native-base'

export function Example() {
  return (
    <>
      {/* Text badge */}
      <Badge>New</Badge>

      {/* Badge with expression */}
      <Badge>{count}</Badge>

      {/* Badge with colorScheme (maps to state) */}
      <Badge colorScheme="success">Success</Badge>

      {/* Badge with size and transparent */}
      <Badge size="lg" transparent>
        Large
      </Badge>

      {/* Indicator dot pattern (no text, just styles) */}
      <Badge
        rounded="full"
        p="0"
        w="12px"
        h="12px"
        bg="icon.brand"
        position="absolute"
        top="0"
        right="0"
      />

      {/* Badge with testID */}
      <Badge testID="status-badge">Active</Badge>

      {/* Badge with layout props */}
      <Badge mt="sm" ml="xs">
        Beta
      </Badge>
    </>
  )
}
"
`;
